Shared/Characters/GhostBase.gd

```diff
@@ -1,31 +1,45 @@
 extends CharacterBase
 class_name GhostBase
 
-
+var _is_active = false
 var _is_playing: bool = false
-var _record_data: RecordData
+var _record_data: RecordData = null
 var _current_frame_index: int = -1
 var _start_time: int = -1
 
 
-func ghost_base_init(action_manager, record_data: RecordData) -> void:
-	_record_data = RecordData.new().copy(record_data)
-	self.timeline_index = record_data.timeline_index
+func init(action_manager, player_id, team_id, timeline_index, spawn_point) -> void:
+	self.player_id = player_id
+	self.team_id = team_id
+	self.timeline_index = timeline_index
+	self.spawn_point = spawn_point.global_transform.origin
 	.character_base_init(action_manager)
 
+func set_record_data(record_data):
+	_record_data = RecordData.new().copy(record_data)
 
-# OVERRIDE #
-func reset() -> void:
-	.reset()
-	_is_playing = false
-	_current_frame_index = -1
-	_start_time = -1
+func clear_record_data():
+	_record_data = null
+
+func is_active() -> bool:
+	return _is_active
+func is_playing() -> bool:
+	return _is_playing
 
+func is_record_data_set() -> bool:
+	if _record_data:
+		return true
+	else:
+		return false
 
 # Goes through the record and applies each frame
-func _physics_process(_delta):
+func update(_delta):
+	if not _is_active:
+		return
 	if not _is_playing:
 		return
+	if not is_record_data_set():
```

Comment:
Yeah I guess you're right.  As far as I could tell this if is unnecessary anyway as `_is_active` only ever true after records are set. Instead of logging I'll use an assert, so we don't have to think about removing it when building.